Terminal.Gui/Views/CheckBox.cs

```diff
@@ -11,8 +11,7 @@ public class CheckBox : View
     private bool? _checked = false;
```

Comment:
Hmm. Why is this nullable? For tri-state?

Reply:
Yes.

Reply:
Ok cool.

Suggestion, then: Use an enum, like how WPF does it.

The main reason is that a nullable boolean (`Nullable<bool>`) is more cumbersome to work with than an enum, and is also more costly (cost is a fairly minor concern though).

Here is the source code for that enum as it exists in WPF (except I removed the ` : byte` because that's not helpful for us):

```csharp
    /// <summary>
    /// Visibility - Enum which describes 3 possible visibility options.
    /// </summary>
    public enum Visibility
    {
        /// <summary>
        /// Normally visible.
        /// </summary>
        Visible = 0,
 
        /// <summary>
        /// Occupies space in the layout, but is not visible (completely transparent).
        /// </summary>
        Hidden,
 
        /// <summary>
        /// Not visible and does not occupy any space in layout, as if it doesn't exist.
        /// </summary>
        Collapsed
    }
```

Optional of course.

Reply:
But the `_checked` field is for a tri-state `Boolean` and not for the `Visibility`. It only indicate if a `CheckBox` is `true`, `false` or `null` indicating that is not defined yet.

Reply:
Oh whoops yes sorry I don't know why my head went to Visibility out of nowhere. 😅 

There's also a nice property in WPF for Checked which is `IsThreeState`. It defaults to false, and the checked boolean is exactly as you have it - a nullable bool - and its default value is false as well.

Again - totally optional, though a nice and expressive property.

Reply:
Yes, an enum could actually be used for this. It's up to @tig :-)

Reply:
Well _this_ one is a `bool?` in WPF, so I'm not interested in that part any more. Sorry I wasn't clear.

The only remaining very optional suggestion I had for this part was adding in that `IsThreeState` bool.

Reply:
Just for clarity:

IsThreeState isn't simply informational. It changes the behavior of a checkbox for whether it will _allow_ setting null at runtime.

Reply:
But, thinking about that, it's probably more appropriate for that to be in a different PR anyway, if we decide to add it. It's not related to this.